### PR TITLE
Sup 33 review features from js api

### PR DESCRIPTION
**Pull request by adolfo1971**
Hi Norman, this are the changes:

Added missing features that exist in the JS API in the update method and in the list method of the StorageFile library

Regards

### CODE REVIEW

File: src/Util/Request.php [hunk 1]
```diff
@@ -46,6 +29,7 @@ public static function handleError($error)
 			$response = $error->getResponse();
 			$data = json_decode($response->getBody(), true);
 
+			//return $response;
```
Comment (tehnorm) [inline src/Util/Request.php]: Stray comment - we should remove.

File: tests/integration/StorageBucketTest.php [hunk 2]
```diff
@@ -11,8 +11,11 @@ final class StorageBucketTest extends TestCase
 	public function setup(): void
 	{
 		parent::setUp();
-		$authHeader = ['Authorization' => 'Bearer '.'<service_role>'];
-		$this->client = new  \Supabase\Storage\StorageClient('https://<project_ref>.supabase.co', $authHeader);
+		$dotenv = \Dotenv\Dotenv::createUnsafeImmutable(dirname(__DIR__, 2), '.env.test');
```
Comment (tehnorm) [inline tests/integration/StorageBucketTest.php]: We should use a path here e.g. `__DIR__ . '/../../env.test';` vs `dirname()`

File: tests/integration/StorageFileTest.php [hunk 3]
```diff
@@ -14,10 +14,12 @@ final class StorageFileTest extends TestCase
 	public function setup(): void
 	{
 		parent::setUp();
-		$authHeader = ['Authorization' => 'Bearer '.'<service_role>'];
-		$bucket_id = '<existing-storage-bucket>';
-		$this->client = new  \Supabase\Storage\StorageFile('https://<project_ref>.supabase.co/storage/v1',
-			$authHeader, $bucket_id);
+		$dotenv = \Dotenv\Dotenv::createUnsafeImmutable(dirname(__DIR__, 2), '.env.test');
```
Comment (tehnorm) [inline tests/integration/StorageFileTest.php]: We should use a path here e.g. __DIR__ . '/../../env.test'; vs dirname()

File: tests/unit/VCRStorageBucketTest.php [hunk 4]
```diff
@@ -11,8 +11,11 @@ class VCRStorageBucketTest extends TestCase
 	public function setup(): void
 	{
 		parent::setUp();
-		$authHeader = ['Authorization' => "Bearer {$_ENV['SERVICE_ROLE']}"];
-		$this->client = new  \Supabase\Storage\StorageClient("https://{$_ENV['PROJECT_REF']}.supabase.co/storage/v1", $authHeader);
+		$dotenv = \Dotenv\Dotenv::createUnsafeImmutable(dirname(__DIR__, 2), '.env.test');
```
Comment (tehnorm) [inline tests/unit/VCRStorageBucketTest.php]: We should use a path here e.g. __DIR__ . '/../../env.test'; vs dirname()

File: tests/unit/VCRStorageFileTest.php [hunk 5]
```diff
@@ -14,10 +14,12 @@ final class VCRStorageFileTest extends TestCase
 	public function setup(): void
 	{
 		parent::setUp();
-		$authHeader = ['Authorization' => 'Bearer '.$_ENV['SERVICE_ROLE']];
+		$dotenv = \Dotenv\Dotenv::createUnsafeImmutable(dirname(__DIR__, 2), '.env.test');
```
Comment (tehnorm) [inline tests/unit/VCRStorageFileTest.php]: We should use a path here e.g. __DIR__ . '/../../env.test'; vs dirname()

File: .env.test [hunk 6]
```diff
@@ -0,0 +1,2 @@
+REFERENCE_ID=gpdefvsxamnscceccczu
```
Comment (tehnorm) [inline .env.test]: We should not check in the values here - just the keys - this is a security issue.

**Review by tehnorm — CHANGES_REQUESTED** [review]
Good work. Few items to adjust and this should be good to go. Let me know if you have questions.

File: src/Util/Request.php (hunk 1 above)
Comment (adolfo1971) [reply src/Util/Request.php]: Done

File: tests/integration/StorageBucketTest.php (hunk 2 above)
Comment (adolfo1971) [reply tests/integration/StorageBucketTest.php]: Done

File: tests/integration/StorageFileTest.php (hunk 3 above)
Comment (adolfo1971) [reply tests/integration/StorageFileTest.php]: Done

File: tests/unit/VCRStorageBucketTest.php (hunk 4 above)
Comment (adolfo1971) [reply tests/unit/VCRStorageBucketTest.php]: Done

File: .env.test (hunk 6 above)
Comment (adolfo1971) [reply .env.test]: Ok 
I did it without checking, I'll be careful in the next ones

File: tests/unit/VCRStorageFileTest.php (hunk 5 above)
Comment (adolfo1971) [reply tests/unit/VCRStorageFileTest.php]: Done

**Comment by adolfo1971** [comment]
Hi @tehnorm 
Today Continue including missing features in Storage File Class create Signed Url,
getPublicUrl and upload and the points commented in the last revision were modified.

A detail, when testing the get-public-url-file-with-transformations method returns an error, and it seems that it is permissions from supabase is this:

{"statusCode":"403","error":"FeatureNotEnabled","message":"feature not enabled for this tenant"}

Regards

**Pull request merged**